File: backend/src/pomelo_orbit/interfaces/api/project/dto.py
from datetime import datetime

from pydantic import BaseModel, Field

from pomelo_orbit.domain.auth.entities import User


class ProjectCreateReq(BaseModel):
    name: str = Field(min_length=1, max_length=100)
    code: str = Field(min_length=1, max_length=100, pattern=r"^[a-z0-9_-]+$")


class ProjectUpdateReq(BaseModel):
    name: str = Field(min_length=1, max_length=100)
    code: str = Field(min_length=1, max_length=100, pattern=r"^[a-z0-9_-]+$")


class ProjectResp(BaseModel):
    id: str
    name: str
    code: str
    is_active: bool
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class ProjectMemberReq(BaseModel):
    user_id: str = Field(min_length=1)


class ProjectMemberResp(BaseModel):
    id: str
    username: str
    email: str | None

    @classmethod
    def from_domain(cls, user: User) -> "ProjectMemberResp":
        return cls(id=user.id, username=user.username, email=user.email)

File: internal/transport/http/handler/project/dto.go
package projecthandler

import transportresponse "backend/internal/transport/http/response"

type ProjectResp struct {
	Id        string `json:"id"`
	Name      string `json:"name"`
	Code      string `json:"code"`
	IsActive  bool   `json:"is_active"`
	CreatedAt string `json:"created_at"`
	UpdatedAt string `json:"updated_at"`
}

type ProjectMemberResp struct {
	Id          string  `json:"id"`
	Username    string  `json:"username"`
	Email       *string `json:"email"`
	Status      string  `json:"status"`
	AuthSource  string  `json:"auth_source"`
	LastLoginAt *string `json:"last_login_at"`
}

type ProjectListResp = transportresponse.ListResp[ProjectResp]
type ProjectMemberListResp = transportresponse.ListResp[ProjectMemberResp]

type ProjectSaveReq struct {
	Name string `json:"name"`
	Code string `json:"code"`
}

type ProjectMemberReq struct {
	UserId string `json:"user_id"`
}
